.github/workflows/publish-sources.yml

```diff
@@ -0,0 +1,88 @@
+name: Publish
+
+on:
+  push:
+    branches: # Snapshots
+      - main
+
+jobs:
+  publish-sources:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Package sources
+        run: |
+          ./sources.sh
+      - name: Publish nightly sources
+        uses: softprops/action-gh-release@v1
+        with:
+          prerelease: true
+          name: nightly
+          tag_name: nightly
```

Comment:
@mkurz Needs to choice a name of tag for `main` branch. WDYT?

Reply:
You mean if the tag name `nightly` is ok? yes I am ok with it.
However, will the tag be overriden each time pushing to main? (which should be)

Reply:
> You mean if the tag name `nightly` is ok? yes I am ok with it. 

Ok. 

> However, will the tag be overriden each time pushing to main? (which should be)

Good question 🤔 I think yes, but need to check. I've checked only overriding artifacts and it works as expected. After any merge/push to `main`/`3.0.x`/`2.9.x`/etc branches an archives with sources will override.

Reply:
> The current default branch is `3.0.x`. I think we should set this property explicitly for any branches which sources we want to publish. WDYT?

Yes, absolutely. We _need_ to set
```
target_commitish: main
```
in this pull request.